plugins/action/win_updates.py

```diff
@@ -831,7 +835,17 @@ def _run_sync(self, task_vars, module_options, reboot, reboot_timeout):  # type:
             round += 1
             display.v("Running win_updates - round %s" % round, host=task_vars.get('inventory_hostname', None))
 
-            update_result = self._run_updates(task_vars, module_options, poll_script_path, cancel_script_path)
+            try:
+                update_result = self._run_updates(task_vars, module_options, poll_script_path, cancel_script_path)
+            except _RecreateTempPathException:
+                display.vv("Failure when running win_updates module with existing tempdir, retrying with new dir")
+                self._connection._shell.tmpdir = None
+                self._make_tmp_path()
+                module_options['_output_path'] = self._connection._shell.tmpdir
+                poll_script_path = self._copy_script(_POLL_SCRIPT, 'poll.ps1')
+                cancel_script_path = self._copy_script(_CANCEL_SCRIPT, 'cancel.ps1')
```

Comment:
Only nit here is it feels wrong to duplicate the creation vs just making it a function, otherwise seems pretty reasonable

Reply:
True, it does mean I have to return 3 values but it makes sense to try and collect all the operations for setting up a tempdir for use in the module in one location.